docs/editing-components.md

```diff
@@ -0,0 +1,376 @@
+   into a single unit (what we'd think of as a "module" in the Wagtail editor).
+   This one has three sub-blocks (lines 2, 3, and 4).
+2. The `heading` block uses the basic Wagtail `CharBlock`, which results in a
+   field with a basic single-line text input.
+3. The `paragraph` block uses the basic Wagtail `RichTextBlock`,
+   which results in a field with a multiline WYSIWYG text input.
+4. The `links` block uses another basic Wagtail block, `ListBlock`,
+   which is a special type of block that can hold a variable number of
+   some other block (the `Hyperlink` atom block, in this case).
+5. The `Meta` class defines some properties on the `RelatedContent` block
+   that are used by the Wagtail admin or in rendering the block.
+6. The `icon` property tells Wagtail what icon to use in the editor
+   for the button you use to add a `RelatedContent` block to a StreamField.
+   Icon options can be found in the Wagtail style guide when running locally:
+   http://localhost:8000/admin/styleguide/#icons
+7. The optional `label` property overrides the text of that same button;
+   if `label` is not set, Wagtail will generate one from the name of the block.
+8. The `template` property is a pointer to the HTML template used to render
+   this component. [See below for more on templates.](#the-html-template)
+
+
+### Adding it to a StreamField
+
+Components are made available in the page editing interface
+by adding them to one of a page types's StreamFields.
+These are usually the first things in a page's class definition.
+For example, see this snippet from `blog_page.py`:
+
+```python
+class BlogPage(AbstractFilterPage):
+    content = StreamField([
+        ('full_width_text', organisms.FullWidthText()),
+        ('info_unit_group', organisms.InfoUnitGroup()),
+        ('expandable', organisms.Expandable()),
+        ('well', organisms.Well()),
+        ('email_signup', organisms.EmailSignUp()),
+        ('feedback', v1_blocks.Feedback()),
+        ('image_text_50_50_group', organisms.ImageText5050Group()),
+    ])
+    …
+```
+
+This sets up a StreamField named `content` that allows for
+the insertion of any of those seven listed blocks into it.
+
+To make the `RelatedContent` module (shown above) available to this StreamField,
+we'd add a new entry to this list following the same format:
+`('related_content', molecules.RelatedContent()),`.
+
+Most page types have a couple StreamFields (`header` and `content`) in the
+general content area (the first tab on an editing screen), and most also share
+[a common `sidefoot` StreamField](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/models/base.py#L95-L107)
+(so named for the fact that it appears on the right side on some page types,
+but in the footer on others) on the sidebar tab.
+
+
+---
+
+!!! note "Warning: Migrations may be required!"
+    Changes to either of the Python bits will usually require
+    a new schema migration file, and _may_ require a data migration.
```

Comment:
"Python bits"? 😛 

I think this note creates more mystery and more uncertainty than necessary. It can say conclusively something like:

> Adding or changing fields on either Python class will require a new [Django schema migration](https://docs.djangoproject.com/en/1.11/topics/migrations/), and changing fields on a pre-existing block will require a [data migration](https://docs.djangoproject.com/en/1.11/topics/migrations/#data-migrations). 

At least until Wagtail can handle changes to StreamFields a bit better.

You could even explain a little bit of why (that changes to the block appear as changes to the page model to the Django migration system), but maybe that's too much info, or better below. I just don't think migrations and when they need to happen should be mysterious.

Reply:
Addressed in a47039c